src/types/UIProtocol.ts

```diff
@@ -13,13 +13,23 @@ export enum ProtocolVersion {
   '0.0.1' = '0.0.1',
 }
 
-export enum ProtocolCommand {
+export enum MessageCode {
+  REQUEST = 0,
+  ANSWER = 1,
+  ERROR = 2,
+}
+
+export type ProtocolMessage = [MessageCode, ...ProtocolRequest];
+
+export type ProtocolRequest = [number, ...ProtocolCommand];
```

Comment:
You can use ProtocolMessage everywhere, the protocol complete payload is expected to be the same for request and response.